### DISCUSSION TITLE
Sujet / Thema 10 –– Standards und Werkzeuge / Standards et outils

### DISCUSSION THREAD

**Discussion by rastrau** [GeoUnconference/discussions#8] [Ideen / Idées]
Resultate der Diskussion an der GeoUnconference 1 / Résultats de la discussion lors de la GeoUnconference 1

![Standards, interoperability, and web services](https://user-images.githubusercontent.com/4510815/124478836-e23e2d00-dda5-11eb-8657-6172594f3615.jpg)

![WebGIS, interoperability, OGC, and vector tiles](https://user-images.githubusercontent.com/4510815/124478843-e5d1b400-dda5-11eb-9e1c-e4d9ff472511.jpg)

**Comment by loleg** [comment]
Outputs aus unseren weitergezogenen Diskussion am Geounconference #2 (Andry, Filli, Patricia, Oleg)

![Group8-MetadataStandards](https://user-images.githubusercontent.com/31819/132707665-13181ce1-708d-47d7-977e-47a8fd375fed.png)

**Reply by loleg** [reply]
![IMG_20210909_161208](https://user-images.githubusercontent.com/31819/132705169-255893b5-622f-471e-9acf-4178fc8462fe.jpg)

**Reply by piMoll** [reply]
Das QGIS Plugin "Swiss Geo Downloader" kann direkt im QGIS Plugin Manager heruntergeladen werden. Weitere Infos zum Plugin und der STAC API sind in der [Plugin Doku](https://pimoll.github.io/swissgeodownloader/) zu finden.

**Reply by ajoos** [reply]
Die wichtigsten Links aus obigem Bild noch klickbar:
* https://plugins.qgis.org/plugins/swissgeodownloader/
* https://data.geo.admin.ch/
* https://pygeoapi.io/
* https://ech.ch/index.php/de/standards/60396

_Danke für den interessanten & regen Austausch und Workshop, liebe Gruppemitglieder!_

**Reply by loleg** [reply]
*Ein grosses Merci an euch!* Here are a couple of other links we looked at together:

- The [quick intro](https://stacspec.org/overview.html) and more in-depth [STAC Overview](https://github.com/radiantearth/stac-spec/blob/master/overview.md#stac-overview) documents.
- [MeshTiler](https://dribdat.cividi.space/project/51) as a project whose author specifically [called out](https://twitter.com/rkaravia/status/1368608865588285440) the STAC API as beneficial in development.
- [STAC Documentation](https://www.geo.admin.ch/en/geo-services/geo-services/download-services/stac-api.html) at GeoAdmin
- https://geostandards.ch/
- ["What is the difference in purpose between STAC and OGC services?"](https://gis.stackexchange.com/questions/358609/what-is-the-difference-in-purpose-between-stac-and-ogc-services) - see also this [post by Chris Holmes](https://medium.com/radiant-earth-insights/spatiotemporal-asset-catalogs-and-the-open-geospatial-consortium-659538dce5c7?source=user_profile---------8----------------------------)
- [Implementing STAC in CKAN](https://github.com/ckan/ckan/discussions/6364)
- [xkcd: standards](https://xkcd.com/927/)

**Reply by rastrau** [reply]
Super, vielen Dank!

**Reply by maxcollombin** [reply]
### How to choose the right standards for the NGDI ?

#### [GeoUnconference #3](https://ti.to/geounconference/3): Elfie, Bertil, Frank, Jens, Maxime

#### 4 « W » – « who, what, where, and why ? »

- Who:      Geodata users
- Where:    Switzerland
- What:     Geodata
- Why:      Ensuring simple use of geodata to create value

#### Evaluation criteria

|Standard|User needs and constraints|Legal framework|Simple|Understandable|Easy to implement|Sustainability|Open|
|:-----:|:-----:|:-----:|:-----:|:-----:|:-----:|:-----:|:-----:|
|WxS|+/-|+|+/-|+/-|+|+|+|
|OGC APIs|+|?|+|+|+|+|+|
|MVT|+/-|-|+|+/-|+|-|+/-|
|INTERLIS|+|+|-|-|+/-|?|+|

**Reply by rastrau** [reply]
I've cross-posted these results also in the discussion forum of the INTERLIS user meetups: https://github.com/moflexch/ili-anwendertreffen/discussions/9. 

@maxcollombin: Au cours de la présentation de vos résultats, tu as fait quelques remarques précieuses sur les raisons pour lesquelles vous avez choisi ces critères comme étant pertinents et aussi un peu plus sur ce que le groupe entendait exactement par "sustainability" et les autres termes. Pourrais-tu les expliquer brièvement ici afin de favoriser une discussion plus approfondie ?

**Reply by maxcollombin** [reply]
Hi @rastrau,

Thank you for your interest in our proposal.
I am happy to specify the criteria.

#### User needs and constraints:
    
First of all, it is important to note that there are different user profiles with different needs.
This list could of course be further specified, as could the user profiles (population, decision-makers, GIS specialists, ...) and their specific needs.
Without going into too much detail, we had thought of the following parameters:

- Mutlilingualism
- CRS
- Tile Matrix Set
- XML/JSON Schema

#### Legal framework:

Standards play an important role in the NGDI.
In order to ensure their quality and interoperability, guidelines and recommendations are required.
This is the case, for example, with the eCH-0056 standard, which is explicitly mentioned in [Art. 7 of the OGéo-swisstopo](https://www.fedlex.admin.ch/eli/cc/2008/390/fr#art_7).

#### Simple:

In order to promote its adoption, a specification must first and foremost be simple.
(e.g. modular and recurring components of the **REST** APIs)

#### Understandable:

In order to guarantee its use, a standard must be understandable to the widest possible community (e.g. web community).

#### Easy to implement[^1]
 
In order to ensure its integration into a national (geo-)data infrastructure and related developments, the ease of implementation of a specification plays an important role. The international standards are of interest here, as they promote the reproducibility of their implementation (wider community, tutorials, cookbooks).
In this context. the integration of the standard in the target product must be possible to do in an easy way (low coding) so as to be able to promote the reuse of data and associated services
   
#### Sustainability:

The sustainability of a standard refers to the organisation that maintains it. Is this entity neutral? How is it organised? How big is it and what resources does it have? Will it be able to maintain the specification in the long term?

#### Open:

Open means that you can contribute to its evolution.

These are the details that I can currently provide. Please feel free to comment or clarify.

#### Interoperable[^1]

Interoperability is the ability to work with other existing or future products or systems without restriction of access or implementation

[^1]: modified after @schenardi 's [contribution](https://github.com/GeoUnconference/discussions/discussions/8#discussioncomment-2708999)

**Reply by schenardi** [reply]
Thanks @maxcollombin for the details.
Concerning "How to choose the right standards for the NGDI ?" and "easy to implement" a few thoughts from my perspective :

**What is a standard?**
In the discussion through all 3 geo unconferences and in general we often use "OGC" as the one and only standards. De facto it's a good approach to standardize a lot of things - but in practice it's very slow and old and does not support some use cases/requirements. The consequence is that suppliers implement their own standards like e.g. Esri does due to lack of options (probably not only but that happens). One example is the possibility to create and use vector tiles in LV95, or their featureservice with much more capabilities then WFS-T. All the Esri Apps consume basemaps and featureservices to be able to provide the whole functionality. WFS wouldn't be good enough (e.g. for offline-Workflows, large datasets, attachment handling etc). 

De facto, in Switzerland their is a large Esri community (probably not even as large as the open source community, but large enough) and they build another "quasi standard" and want to be able to use the NGDI in the same way as the OGC community. I know that's a "hen egg issue" and the suppliers should adopt to standards - but they cannot not always adopt. And thererfore the discussion should also point in the direction if a NGDI must provide a minimum set of data as well in Esri formats and not only in OGC. 

Minimum datasets from my perpectice to be published on ArcGIS Online as publicly available datasets.
* Amtliche Vermessung as Vector Tiles, offline capable
* Topobasemap (currenty done by Esri CH, Marianne Rüegsegger) as Vector Tiles, offline capable
* Swissimage LV95 as Image Service (which allows dynamic queries, printing, colorchanges, band-selection etc)
* 3D Buildings as i3s (Esri 3D Service)
* Ground as i3s

I'm open to demo some of those datasets (including Amtliche Vermessung) on Laixo's ArcGIS Online account. 

**easy to implement**
Easy to implement is for me > just use it. I don't want to write code, download data and re-publish it etc. I want to use services and more directly in my applications without carying to much about symbology, dataflows etc.

**Reply by maxcollombin** [reply]
Hi @schenardi,

First of all I would like to thank you for your comments.

I agree with you that the notion of standard should not be exclusively limited to the OGC.

Your remark about the fact that the generation of OGC WxS standards 
>[...] does not support some use cases/requirements
is relevant.

This is the reason why OGC is moving towards a [new generation of standards](https://ogcapi.ogc.org/) of which the OGC API Tiles is part
and which should make it possible to [define a tiling scheme from any projection system](http://docs.ogc.org/DRAFTS/20-057.html#_tiling_scheme) as you mention for LV95.
(Although this is only a draft specificiation, solutions such as Geoserver already offer an [early implementation](https://docs.geoserver.org/latest/en/user/community/ogc-api/tiles/index.html)).

Your point about OGC processes being slow and not respecting certain requirements leading
>[...] the suppliers to implement their own standards
is also legitimate.

I think however that one of the key criteria of a good standard (especially in the context of an infrastructure such as the NGDI)
is **interoperability** (to be added to the list😉✅) and the software suppliers should work together towards this goal, regardless of whether the solutions they propose are proprietary or open:
This is the case for the [OGC API Tiles](http://docs.ogc.org/DRAFTS/20-057.html) (iv. Submitting organizations) and [I3S](https://docs.ogc.org/cs/17-014r8/17-014r8.html) (iv. Submitting organizations).

Your comment
>De facto, in Switzerland their is a large Esri community[...]

reminds me of the discussion: [Sujet / Thema 7 –– Let's map the NGDI! 🚀](https://github.com/GeoUnconference/discussions/discussions/31)
in which I would find it interesting to map the solutions mainly used in Switzerland (ESRI, MapServer, QGIS Server, Geonetwork,...) to make sure that their communities are well represented in the organizations promulgating standards.
What would be your opinion on this @rastrau?

I think your demo could be a good idea in that it would provide a good example of "User needs and constraints".

I will also add your comment regarding the "Easy to implement" aspect✅.

This discussion provides us with very interesting inputs for our [NGDI-20-60 project](https://www.interlis.ch/veranstaltungen/bisherige-veranstaltungen/spirgarten-2022) so thanks a lot!🙏🏻

**Reply by rastrau** [reply]
(Sorry, holiday hiatus)

To the first point @schenardi touches upon («What is a standard?»), I think: We all know there are *de iure* and *de facto* standards. The former are usually equated with the [Open Geospatial Consortium OGC](https://www.ogc.org/) standards (globally) – as pointed out by @schenardi – and the [eCH](https://www.ech.ch/) standards (nationally) for the GIS industry. The latter (de facto standards) usually originate with vendors or maybe (proponents of) specific communities. Here, e.g. [Indexed 3D Scene Layers I3S](https://github.com/Esri/i3s-spec) come to mind, or Esri File Geodatabase and Esri Shapefile, historically,

With regards to standardization being «very slow»: I think that’s true. But I would give OGC credit for setting up, and want to highlight, the [OGC Community Standards process](https://www.ogc.org/standards/community) which in my understanding is aimed exactly at capturing «emerging» standards that originate outside of the traditional and slow OGC standardization process. The above-mentioned [I3S](https://www.ogc.org/standards/i3s) is such an OGC Community Standard (same as [3D Tiles](https://www.ogc.org/standards/3DTiles), by the way, to stay within the same application domain).

I would like the [#SwissGIS](https://twitter.com/search?q=%23SwissGIS) community to adopt even more OGC Standards, Community and other, and thus to promote **interoperability** even more than today (as highlighted by @maxcollombin). I do not think public administration should have an *obligation* to offer formats and services that are not *in some way* standardized, one of these ways (not the only one) being the OGC Community Standard route.

My «2 cents» regarding the «easy to implement» point by @schenardi (« I don't want to write code, download data and re-publish it etc. I want to use services and more directly in my applications without carying to much about symbology, dataflows etc.»). This perspective is fair enough and I sometimes also incorporate it. However, I can also see use cases where a user does want to access bulk data, e.g. under a stable URL. This kind of offer is embodied e.g. by [data.geo.admin.ch](https://data.geo.admin.ch). 

Turns out, I’ve had [discussions about this point several times](https://twitter.com/search?q=from%3Arastrau%20bulk) since 2017. E.g:

[![image](https://user-images.githubusercontent.com/4510815/171574458-a1be1020-649f-4b7d-a833-3c1585ad6214.png)](https://twitter.com/rastrau/status/824282903421054976)

and

[![image](https://user-images.githubusercontent.com/4510815/171574415-d82454d2-e581-4b40-a0d2-9ddbc11fc38c.png)](https://twitter.com/rastrau/status/1528745434608455681)

By the way, the linked article from the first tweet states (highlighted by myself): «**A data API alone, if it does not provide bulk data, does not meet the principles of open government data**. If there is a choice between bulk data or a data API which does not provide bulk data, bulk data must come first.» I agree.

@maxcollombin: I share your interest regarding an analysis of the technical underpinnings of the NGDI/INDG. An overview of which solutions are employed in how many components of the NGDI/INDG and where would indeed be an interesting product and foundation for further discussion and evolution of the NGDI/INDG.

I will be interested in the results of the **[NGDI-20-60 project](https://www.geo.admin.ch/content/geo-internet/de/about-swiss-geoportal/task-and-responsabilities/ngdi-funds/_jcr_content/contentPar/accordion/accordionItems/346_1589792279400/accordionPar/downloadlist_copy_co/downloadItems/153_1598904341785.download/20-60_ngdi_entscheid.pdf)** (pdf) you mentioned, @maxcollombin. **Who participates in that?**

If you agree that public administration in Switzerland should base their offerings on technologies that *meet some kind of standard*, **what do you regard as relevant / "valid" standardization bodies in the GIS domain, besides the already mentioned OGC and eCH?**

(Maybe this discussion is of interest also to @edigonzales, @davidoesch and @gjn, as we've had similar conversations in the past.)

**Reply by p1d1d1** [reply]
Interesting discussion but the comparison is mixing stuff: service specifications and data modeling language/data format. Comparing e.g. OGC API and INTERLIS doesn't make too much sense IMHO. Complexity is not (only) in the standard but in the data model. If you data model is complex then it's implementation will be in any case complex.

**Reply by edigonzales** [reply]
Assumption: Most people do not expect or "see" a data model when they just download some flat table like data as gpkg or shapefiles.

**Reply by p1d1d1** [reply]
or like a "flat xtf"

**Reply by rastrau** [reply]
I know of an office that looks into offering a (Interlis) model-based pick-and-choose datashop: You chose topics, classes, attributes of interest and the service exposes the "flattened" data under a dedicated web service endpoint for you.

**Reply by edigonzales** [reply]
Mmmh, heard that too since ages. Always reminds of the "kentucky fried mouse" of "the beach": https://getyarn.io/yarn-clip/541073af-1ad0-4e09-8e5a-7bad11427486 

Joking aside: would be great. Isn't it ConfConv (ConvConf?), is it?

**Reply by p1d1d1** [reply]
IMHO each time an Interlis model is published there should always be an associated "offical" "flat" "view"

**Reply by p1d1d1** [reply]
... or alternatively the "flattening" rules could be defined at the standard level

**Reply by edigonzales** [reply]
So let's start with all those MGDM. E.g. geodienste.ch does it anyway as input for their WMS (though not with INTERLIS). 

And we do this with all of our cantonal data as part of our publication process. There is always an "edit model" and a "publication model". Works like a charm.

**Reply by p1d1d1** [reply]
> And we do this with all of our cantonal data as part of our publication process. There is always an "edit model" and a "publication model". Works like a charm.

Do you publish somewhere your "publication models"?

**Reply by edigonzales** [reply]
https://modelfinder-rgank.ondigitalocean.app/?ilisite=geo.so.ch&expanded=true -> search for "Pub" or "Publikation"
Same here: https://geo.so.ch/models/ilimodels.xml 

We have some rules for Publikationsmodelle:
- no assocations
- only one geometry column per class
- inheritance is allowed as long as it ends up with a reasonable structure (see above) in the database / geopackage  

see: https://sogis.github.io/modellbasierte-datenerfassung-handbuch/#_publikationsmodell

**Reply by maxcollombin** [reply]
👋🏻Hi @rastrau,

I think I'll take the cake for the delay😅🤣.

> I will be interested in the results of the NGDI-20-60 project [...]. **Who participates in that?**

The INDG 20-60 project is part of the INDG's dedicated resources and is currently coordinated by Frank Gottsmann with the participation of the HEIG-VD, the UNIGE and the SUPSI.

It aims at following the design of the new OGC standards in order to propose recommendations for the NGDI. 

As you may know, an initiative coordinated by [Geostandards.ch](https://geostandards.ch/) has been launched to start the revision of the [eCH-0056 standard](https://www.ech.ch/de/standards/60396). We hope that the results of this project will be useful for it and we encourage you to participate in the upcoming workshops. For further information, please contact the team [Geostandards.ch](https://geostandards.ch/) ([F.Gottsmann or P. Staub](standardisierung@geostandards.ch))
    
> If you agree that public administration in Switzerland should base their offerings on technologies that _meet some kind of standard_, **what do you regard as relevant / "valid" standardization bodies in the GIS domain, besides the already mentioned OGC and eCH?**

Depending on what you consider behind the words **relevant/valid**, in addition to eCH and OGC, we may also mention:

- The joint initiative of the OGC & W3C: ["Spatial Data on the Web"](https://www.w3.org/2015/01/spatial)
- [ISO](https://www.iso.org/standard/54598.html) (International Organization for Standardization which publish [IFC (ISO 16739-1:2018)](https://www.iso.org/standard/70303.html))
- [INSPIRE](https://inspire.ec.europa.eu/) (Infrastructure for Spatial Information in Europe)
- [W3C](https://www.w3.org/standards/geo/) (World Wide Web Consortium) 
- [SIA](https://www.sia.ch/fr/standardisation/standardisation-nationale/standardisation-nationale.html) (Swiss Society of Engineers and Architects which raises an interest to, for example, industrial service networks)

**Reply by rastrau** [reply]
Hey @maxcollombin 

🙂🙂 Thanks a lot for these answers tho! That's a good collection of standards / standards bodies! I'll watch for the Geostandards.ch workshops.

**Comment by rastrau** [comment]
In GeoUnconference 3, a group including @ebelo has discussed about geo-services. Their notes / sketch is embedded below. An important take-away message for me was, that "traditional" OGC services should be accompanied by providing new innovative cloud-native / stream-able data formats such as:
- [COG (Cloud-Optimized GeoTiff)](https://www.cogeo.org/), 
- [COPC (Cloud-Optimized Point Cloud)](https://copc.io/), 
- [Geoparquet](https://github.com/opengeospatial/geoparquet), and 
- [FlatGeoBuf](https://github.com/flatgeobuf/flatgeobuf)

![1](https://user-images.githubusercontent.com/4510815/170707613-145419d3-8bfb-4d4d-a7cb-442994eae0e1.jpg)
.

**Comment by olivergrimm** [comment]
after reading all your great posts from the last months I suggest for tomorrows GeoUnconference 4 to focus on the following themes:

- we write down a classified set of modern, priority use cases relating the use of geodata. imho: this use cases should be independant from any tech or system aspects.
- we try to conclude the list of relevant standards (and probably formats?) based on that set. Therefore we include the tech/system aspects to have a practice useable, state of the art collection.
- we assign the major outputs from the further sessions to that list. This assignment could contain technology aspects, samples, tools, links to activities (like the eCH-0056 revision) etc.

Our product tomorrow evening could probably be a markdown-file containing those concluding information.

Is that a plan?

**Reply by gperonato** [reply]
Sorry I registered last minute to the conference and I am new here. That sounds a good plan!
I have seen that in the previous posts there were some mentions on 3D compatibility. If this is the right place/moment, I'd like to discuss about possibilities to access 3D building data (i.e. the new swissBUILDINGS3D) as a web-service and a possible format (e.g. CityGML and/or CityJSON).
Looking forward to tomorrow's workshop!
Giuseppe

**Reply by davidoesch** [reply]
3d data can be accessed via https://api.geo.admin.ch/services/sdiservices.html#d-tiles and https://api.geo.admin.ch/services/sdiservices.html#terrain-service

**Reply by gperonato** [reply]
I meant building 3d data. Is swissBUILDINGS3D already available? I see only a mention of partial coverage in the 3d tiles but no official mention in the available datasets.

**Reply by davidoesch** [reply]
All swissbuildings in 3D are available as service , covering all of Switzerland

**Reply by maxcollombin** [reply]
I totally agree with the idea proposed by @olivergrimm 💡🚀.

The table below represents a first result of the kick-off meeting of the ecH-0056 revision.

The idea here was to identify needs and use cases and their relation to normative requirements.

|       Use case, data type      | Related services |                                                                                     Standard, Link[^1]                                                                                    | Remark |
|:------------------------------:|------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|--------|
| 3D point clouds                |                  | [COPC](https://copc.io/)                                                                                                                                                                  |        |
| 3D picture                     |                  | [GeoPose](https://github.com/opengeospatial/GeoPose)                                                                                                                                      |        |
| Height reference system        |                  | [LN02](https://www.swisstopo.admin.ch/en/knowledge-facts/surveying-geodesy/reference-frames/local/ln02.html)                                                                              |        |
| Address service, Geocoding API |                  | [Geocoding API](https://www.ogc.org/projects/groups/geocodeapiswg)                                                                                                                        |        |
| Building 3D models             |                  | [IFC](https://www.iso.org/standard/70303.html), [CityGML](https://www.ogc.org/standards/citygml), [CityJSON](https://docs.ogc.org/cs/20-072r2/20-072r2.html)                              |        |
| 3D mesh                        |                  | [OGC WCS](https://docs.opengeospatial.org/is/17-089r1/17-089r1.html), [OGC API Coverages](https://ogcapi.ogc.org/coverages/), [OGC API 3D GeoVolumes](https://ogcapi.ogc.org/geovolumes/) |        |
| Data historization             |                  | ?                                                                                                                                                                                         |        |
| Positioning service (swipos)   |                  | [swipos-GIS/GEO](https://www.swisstopo.admin.ch/en/geodata/geoservices/swipos/swipos-dienste/swipos-gisgeo.html)                                                                          |        |
| Quality                        |                  | [INTERLIS 2](https://www.interlis.ch/en/dokumentation/interlis-2), [eCH-118](https://ech.ch/de/ech/ech-0118/2.0)                                                                          |        |

I hope this will be useful as a basis for your work.

[^1]: this list can of course be extended

**Reply by davidoesch** [reply]
Don't forget the cloudnative data formats . Cogtiff geoparquet flatgeobuff zarr ...

**Reply by ajoos** [reply]
Thank you @olivergrimm, your proposal sounds like a good plan to me. Looking forward to meet, learn and contribute.

**Comment by rastrau** [comment]
During GeoUnconference 4, the following results were produced by investigating use cases, the current situation regarding portals, tools, file formats and service types:
https://demo.hedgedoc.org/s/heWv_Tyel

[![image](https://user-images.githubusercontent.com/4510815/196626021-4cba4ee5-05de-47e0-8d5a-f4771198203e.png)](https://demo.hedgedoc.org/s/heWv_Tyel)


The group has also identified four avenues of exploration that would warrant a proof-of-concept (see the end of the linked document) — potentially funded by the [Zweckgebundene NGDI-Mittel](https://www.geo.admin.ch/de/ueber-geo-admin/leistungsauftrag/zweckgebundene-ngdi-mittel.html) / [Moyens dédiés à l'INDG](https://www.geo.admin.ch/fr/geo-admin-ch/mandat-de-prestations/moyens-dedies-indg.html)?
